Sources/Widgets/Widgets3D/ResliceCursorWidget/index.js

```diff
@@ -399,10 +404,46 @@ function vtkResliceCursorWidget(publicAPI, model) {
     // centered on cursor center.
     const planeSource = computeReslicePlaneOrigin(viewType);
 
+    const defaultViewUp = defaultViewUpFromViewType[viewType];
+    const newNormal = plane.getNormal();
+    const newOrigin = plane.getOrigin();
     // Apply rotation onto plane (i.e. origin, p1, p2)
-    planeSource.setNormal(...plane.getNormal());
+    planeSource.setNormal(...newNormal);
     // TBD: isn't it a no-op ?
-    planeSource.setCenter(...plane.getOrigin());
+    planeSource.setCenter(...newOrigin);
+
+    const rotatedOrig = planeSource.getOrigin();
+    const rotatedP1 = planeSource.getPoint1();
+    const rotatedP2 = planeSource.getPoint2();
+
+    // Compute local view up of transformed plane
+    const rotatedViewUp = vtkMath.subtract(rotatedP2, rotatedOrig, [0, 0, 0]);
+    const rotatedPlane = vtkPlane.newInstance({
+      normal: newNormal,
+      origin: newOrigin,
+    });
+    // Project the default viewup we want to fit on
+    const projectedDefaultViewUp = [0, 0, 0];
+    rotatedPlane.projectVector(defaultViewUp, projectedDefaultViewUp);
+
+    vtkMath.normalize(projectedDefaultViewUp);
+    vtkMath.normalize(rotatedViewUp);
+
+    const rotationAngle = vtkMath.angleBetweenVectors(
+      projectedDefaultViewUp,
+      rotatedViewUp
+    );
+
+    // Compute the new plane points
+    const transform = vtkMatrixBuilder
+      .buildFromRadian()
+      .rotate(rotationAngle, newNormal);
+    transform.apply(rotatedOrig);
+    transform.apply(rotatedP1);
+    transform.apply(rotatedP2);
+    planeSource.setOrigin(rotatedOrig);
+    planeSource.setPoint1(...rotatedP1);
+    planeSource.setPoint2(...rotatedP2);
```

Comment:
it would be great to move all this in a separate function for readability, reusability and testing purpose

Reply:
Additionally, it would be great to document a bit more because it is not very obvious